src/main/scala/com/github/simplesteph/ksm/source/GitLabSourceAcl.scala

```diff
@@ -0,0 +1,91 @@
+package com.github.simplesteph.ksm.source
+
+import java.io.StringReader
+import java.nio.charset.Charset
+import java.util.Base64
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.simplesteph.ksm.parser.AclParser
+import com.typesafe.config.Config
+import org.slf4j.LoggerFactory
+import skinny.http.{HTTP, HTTPException, Request, Response}
+
+import scala.util.Try
+
+class GitLabSourceAcl extends SourceAcl {
+
+  private val log = LoggerFactory.getLogger(classOf[GitLabSourceAcl])
+
+  override val CONFIG_PREFIX: String = "gitlab"  
+  final val REPOID_CONFIG = "repoid"
+  final val FILEPATH_CONFIG = "filepath"
+  final val BRANCH_CONFIG = "branch"
+  final val HOSTNAME_CONFIG = "hostname"  
+  final val ACCESSTOKEN_CONFIG = "accesstoken"
+
+  var lastModified: Option[String] = None
+  val objectMapper = new ObjectMapper()
+  var repoid: String = _
+  var filepath: String = _
+  var branch: String = _
+  var hostname: String = _
+  var accessToken: String = _
+
+  /**
+    * internal config definition for the module
+    */
+  override def configure(config: Config): Unit = {    
+    repoid = config.getString(REPOID_CONFIG)
+    filepath = config.getString(FILEPATH_CONFIG)
+    branch = config.getString(BRANCH_CONFIG)
+    hostname = config.getString(HOSTNAME_CONFIG)
+    accessToken = config.getString(ACCESSTOKEN_CONFIG)
+  }
+
+  override def refresh(aclParser: AclParser): Option[SourceAclResult] = {
+    val url =
+      s"https://$hostname/api/v4/projects/$repoid/repository/files/$filepath?ref=$branch"
+    val request: Request = new Request(url)
+
+    // auth header
+    request.header("PRIVATE-TOKEN", s" $accessToken")
+
+    val response: Response = HTTP.get(request)
```

Comment:
so... I have a problem with that. This is not efficient. 

What I would suggest is 
1) if we don't have any lastModified, indeed do a `get`
2) if we do already have one (from the code you wrote), just issue a `head`. See documentation here: https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository  .
From there you can get the `commit_id` and compare it with the one we have in memory. If they're not different, that means we're good (which is the unused code line 72 - for the 304), and if they're different, that means we have to issue a more expensive `get` to retrieve the content.